src/axom/core/ArrayBase.hpp

```diff
@@ -398,17 +435,28 @@ class ArrayBase
     return static_cast<const ArrayType&>(*this);
   }
 
-  /*!
-    \brief Logical offset to get to the given multidimensional index.
-
-    To get from logical offset to memory offset, multiply by spacing().
-  */
+  //// \brief Memory offset to get to the given multidimensional index.
   AXOM_HOST_DEVICE IndexType offset(const StackArray<IndexType, DIM>& idx) const
   {
     return numerics::dot_product((const IndexType*)idx, m_strides.begin(), DIM);
   }
 
-  /// Logical offset to a slice at the given lower-dimensional index.
+  /*!
+   * \brief Returns the number of elements in memory.
+   *
+   *  offset() will return a value between [0, logicalSize()).
+   */
+  AXOM_HOST_DEVICE IndexType memorySize() const
```

Comment:
Is numElements clearer?

Reply:
Memory size is actually the number of elements times the min stride (or spacing).

Reply:
I'm sticking with `memorySize` for now, but added some documentation to clarify a little -- the distinction I wanted to draw is that `memorySize()` is intended to refer to the range in memory where the elements are, while `size()` refers to the number of actual elements in the `Array` or `ArrayView`.

In practice, we're only using `memorySize()` for the underlying bounds checks in `ArrayBase`, and it's not a public exposed method.